shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -27,6 +28,61 @@ where
     pub ctx: Ctx<'ctx, DB, H, CA>,
 }
 
+impl<'ctx, DB, H, CA> EthBridge<'ctx, DB, H, CA>
+where
+    DB: 'static + ledger_storage::DB + for<'iter> ledger_storage::DBIter<'iter>,
+    H: 'static + StorageHasher,
+    CA: 'static + WasmCacheAccess,
+{
+    /// If the bridge's escrow key was changed, we check
+    /// that the balance increased and that the bridge pool
+    /// VP has been triggered. The bridge pool VP will carry
+    /// out the rest of the checks.
+    fn check_escrow(
+        &self,
+        verifiers: &BTreeSet<Address>,
+    ) -> Result<bool, Error> {
+        let escrow_key = balance_key(&xan(), &super::ADDRESS);
+        let escrow_pre: Amount = if let Ok(Some(bytes)) =
+            self.ctx.read_bytes_pre(&escrow_key)
+        {
+            BorshDeserialize::try_from_slice(bytes.as_slice()).map_err(
+                |_| Error(eyre!("Couldn't deserialize a balance from storage")),
+            )?
+        } else {
+            tracing::debug!(
+                "Could not retrieve the Ethereum bridge VP's balance from \
+                 storage"
+            );
+            return Ok(false);
+        };
+        let escrow_post: Amount =
+            if let Ok(Some(bytes)) = self.ctx.read_bytes_post(&escrow_key) {
+                BorshDeserialize::try_from_slice(bytes.as_slice()).expect(
+                    "Deserializing the balance of the Ethereum bridge VP from \
+                     storage shouldn't fail",
```

Comment:
Good point.

Reply:
yeah, any keys in post should be checked, really. maybe it would be better to disallow reading raw bytes with `read_bytes_post`, since it isn't aware of any concrete types `T` we might be trying to deserialize user data into